### PR TITLE
docs: Make the website's theme adjustable.

### PR DESCRIPTION
**Pull request by PhilipMetzger**
The website will also adjust to the system default theme now.  I mostly did this because I am a darkkmode user and we didn't provide it yet.

### CODE REVIEW

**Comment by ilyagr** [comment]
Seems to work fine for me.

Do you want to update the mkdocs-material version as well? All you need to do is to change a 9.4 to 9.5 in `pyproject.toml` and run `poetry update`.

File: mkdocs.yml
Comment (ilyagr) [inline mkdocs.yml:33]: The name seems swapped between this and the automatic scheme.

File: mkdocs.yml
Comment (ilyagr) [inline mkdocs.yml:22]: (Very minor): Instead of having three comments like this, you could just change line 12 to say something like "...and add a toggle for manually choosing automatic/light/dark palette."

File: mkdocs.yml
Comment (PhilipMetzger) [reply mkdocs.yml:33]: fixed, this was some bad copypasta.

File: mkdocs.yml
Comment (PhilipMetzger) [reply mkdocs.yml:22]: done

**Comment by PhilipMetzger** [comment]
> Do you want to update the mkdocs-material version as well? All you need to do is to change a 9.4 to 9.5 in `pyproject.toml` and run `poetry update`.

I'm generally to lazy to install poetry, so I'll leave it up to you.

**Comment by ilyagr** [comment]
> I'm generally to lazy to install poetry, so I'll leave it up to you.

Sure, I can do that, hopefully very soon. Let's wait with this PR until we merge that then.

**Comment by PhilipMetzger** [comment]
I've adjusted the settings for the new mkdocs version, so this should be good to land.

**Review by ilyagr — APPROVED** [review]
Looks good, thanks!

You might want to manually rebase it onto main so that the PR works as-is, but that will also happen automatically in all that ways that really matter.

**Pull request merged**
